### PR TITLE
feat: add site template

### PR DESCRIPTION
**Pull request by afrodynamic**
# Pull Request Template

## Description

Adds website template for the modules required for the "product detail" page.

## Type of Change

- enhancement: new feature or request

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md) for this project.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Reviewers

@lederius @ika-ika-ika 

## Additional Notes

`npm install` will be necessary to add "react-router-dom" to the project.

### CODE REVIEW

File: client/src/components/ProductDetail/ProductDetail.jsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef } from 'react';
+import { useLocation } from 'react-router-dom';
```
Comment (ika-ika-ika) [inline client/src/components/ProductDetail/ProductDetail.jsx]: You can move useParams and useLocation in the same line 2.

File: client/src/components/pages/ProductDetailPage.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import { NavLink } from 'react-router-dom';
```
Comment (ika-ika-ika) [inline client/src/components/pages/ProductDetailPage.jsx:3]: Move NavLink to line 2.

**Review by ika-ika-ika — CHANGES_REQUESTED** [review]
Please review comments.

File: client/src/components/pages/ProductDetailPage.jsx (hunk 2 above)
Comment (afrodynamic) [reply client/src/components/pages/ProductDetailPage.jsx:3]: Fixed in 5e11f9c

**Review by ika-ika-ika — APPROVED** [review]
Great job!

**Review by lederius — APPROVED** [review]
you know its good

**Pull request merged**